src/webrtc/call.ts

```diff
@@ -631,23 +630,18 @@ export class MatrixCall extends TypedEventEmitter<CallEvent, CallEventHandlerMap
         if (!this.client.getUseE2eForGroupCall()) return;
         // It's possible to want E2EE and yet not have the means to manage E2EE
         // ourselves (for example if the client is a RoomWidgetClient)
-        if (!this.client.isCryptoEnabled()) {
+        if (!this.client.getCrypto()) {
             // All we know is the device ID
-            this.opponentDeviceInfo = new DeviceInfo(this.opponentDeviceId);
+            this.hasOpponentDeviceInfo = true;
             return;
         }
-        // if we've got to this point, we do want to init crypto, so throw if we can't
-        if (!this.client.crypto) throw new Error("Crypto is not initialised.");
-
         const userId = this.invitee || this.getOpponentMember()?.userId;
 
         if (!userId) throw new Error("Couldn't find opponent user ID to init crypto");
 
-        const deviceInfoMap = await this.client.crypto.deviceList.downloadKeys([userId], false);
-        this.opponentDeviceInfo = deviceInfoMap.get(userId)?.get(this.opponentDeviceId);
-        if (this.opponentDeviceInfo === undefined) {
-            throw new GroupCallUnknownDeviceError(userId);
-        }
+        // Here we were calling `MatrixClient.crypto.deviceList.downloadKeys` which is not supported by the rust cryptography.
+        this.hasOpponentDeviceInfo = false;
+        throw new GroupCallUnknownDeviceError(userId);
```

Comment:
We should not leave a codepath always throwing an error. Either we implement this for rust-crypto or we remove this method given it always throws.

Reply:
This is the made with the approval of the @matrix-org/element-call-reviewers https://github.com/matrix-org/matrix-js-sdk/pull/4562 & https://github.com/matrix-org/matrix-js-sdk/pull/4654 & https://matrix.to/#/!EMZCUyCNQMdAjHjiUz:element.io/$pVIKQn97xfrYFdFSkyEzIpXtpPisAjg0juMNSQfJzeI?via=element.io&via=matrix.org&via=t2l.io